### DISCUSSION TITLE
Quicksearch with json data

### DISCUSSION THREAD

**Discussion by TechAspirant** [serenity-is/Serenity#5497] [Help]
Hello,

I am population grid with JSON data using below code
`this.view.setItems([{ Id: 1, Name: "Hello" }, { Id: 2, Name: "World" }], true);`

I want to know how can I override quickserach such that It filters the json data populated in grid and not from the database.

**Comment by TechAspirant** [comment]
Is this the right way ? 

```
protected createQuickSearchInput() {
            Serenity.GridUtils.addQuickSearchInputCustom(this.toolbar.element,
                (field, searchText) => {
                    this.searchText = searchText;
                    this.view.setFilter((item, view) => { return item.Name == this.searchText }); this.view.refresh();
                });
 }
```

**Reply by VictorTomaili** [reply]
if it's works, why not.

**Reply by TechAspirant** [reply]
On clearing search textbox no data appears in the grid.

**Reply by VictorTomaili** [reply]
because there is no item what name is empty string.

**Reply by TechAspirant** [reply]
But on empty search text default content should be loaded in grid.

**Reply by VictorTomaili** [reply]
But you override default filter algorithm with `return item.Name == this.searchText`. you need add condition there for empty search text

**Reply by TechAspirant** [reply]
Is this right and efficient ?
```
        protected createQuickSearchInput() {
            Serenity.GridUtils.addQuickSearchInputCustom(this.toolbar.element,
                (field, searchText) => {
                    this.searchText = searchText;
                    
                    this.view.setFilter((item, view) => {
                        if (this.searchText == '')
                        {
                            return true;
                        }
                        else
                        {
                            return item.Name.toLowerCase().search(this.searchText.toLowerCase()) == -1 ? false : true;
                        }
                    }); this.view.refresh();
                });
        }
```

**Reply by VictorTomaili** [reply]
efficient? I don't know. I didn't tried this or similar anything like this. Right? if it's works, yes it's right. But you can write it more simplified. 

```
this.view.setFilter((item, view) => { 
    return !this.searchText || item.Name.toLowerCase().search(this.searchText.toLowerCase()) == -1;
});
```

**Reply by TechAspirant** [reply]
Shouldn't be != -1. Because it shows remaining data in grid and not the one which matches with search criteria.

**Reply by VictorTomaili** [reply]
yep it's just point to use condition itself instead write true or false.

**Reply by TechAspirant** [reply]
Ok, Thanks for the help.

**Reply by TechAspirant** [reply]
You know why the mouse pointer changes to loading when I hover over search input.
Its not the case with normal search input with SQL data in grid.

**Reply by VictorTomaili** [reply]
Nope I dont know
